.github/workflows/test.yaml

```diff
@@ -0,0 +1,9 @@
+name: Tests
+
+on:
+  pull_request:
```

Comment:
Unless that's implemented in some other way, I would recommend to also run tests on push to main.

The reason for that is if you've got 2 PR's, each may independently pass the tests, but then both are merged, the aggregate may still fail.

Reply:
This was the case previously with the [operator workflows](https://github.com/canonical/operator-workflows) developed by the IS Devops team, but was changed in favor of only running the tests on the PR. PRs will only be merged one at a time to ensure such a case does not happen.